### ISSUE TITLE
Move some settings from app.ini to user settings

### ISSUE THREAD

**Issue by qwerty287** [go-gitea/gitea#17587]
### Feature Description

It would be very nice if some settings could be moved from `app.ini` to the user settings, mainly appearance/UI things. If the user isn't logged in, the default values could be used or the settings in `app.ini` can be kept and used then (i.e. user settings overrides the global settings).
The settings I found that could be moved (maybe some of them are not that good to move them, but they are only affecting the UI or things that can be changed by users, but you have to use a workaround):
* `repository.DEFAULT_BRANCH` (https://github.com/go-gitea/gitea/issues/17236)
* `ui.EXPLORE_PAGING_NUM`
* `ui.ISSUE_PAGING_NUM`
* `ui.MEMBERS_PAGING_NUM`
* `ui.FEED_MAX_COMMIT_NUM`
* `ui.FEED_PAGING_NUM`
* `ui.GRAPH_MAX_COMMIT_NUM`
* `ui.CODE_COMMENT_LINES`
* `ui.MAX_DISPLAY_FILE_SIZE`
* `ui.DEFAULT_SHOW_FULL_NAME`
* `ui.SEARCH_REPO_DESCRIPTION`
* `ui.svg.ENABLE_RENDER`
* `ui.csv.MAX_FILE_SIZE`
* `markdown.ENABLE_HARD_LINE_BREAK_IN_COMMENTS`
* `markdown.ENABLE_HARD_LINE_BREAK_IN_DOCUMENTS`
* `service.ENABLE_USER_HEATMAP`
* `service.AUTO_WATCH_NEW_REPOS`
* `service.AUTO_WATCH_ON_CHANGES`
* `service.explore.DISABLE_USERS_PAGE`
* `api.DEFAULT_PAGING_NUM`
* `time.FORMAT`
* `other.SHOW_FOOTER_BRANDING`
* `other.SHOW_FOOTER_VERSION`
* `other.SHOW_FOOTER_TEMPLATE_LOAD_TIME`

This would help to have less settings in `app.ini` and the possibility for users to change the values if they are not an admin.

**Comment by delvh** [comment]
I disagree with the `ui.*`, `api.*'`, `other.*` and `service.AUTO_WATCH*` settings.
I think for those it is better to leave them globally, especially the limits, as there is a reason only server admins set them.
Otherwise, anyone could basically crash the server by using an insanely high setting, creating a "small" commit/ PR with a few hundred million changes, and then request this file to be loaded. This would basically allow Gitea to **DDOS itself**. I don't want that. For the other things mentioned: I think it is only adding unneeded complexity and confusion because I don't see a reason to customize the footer like that per repo.

So, what I have nothing against is `repository.DEFAULT_BRANCH`, `markdown.*`, `service.ENABLE_USER_HEATMAP` and `time.FORMAT`, and potentially `ui.DEFAULT_SHOW_FULL_NAME`.

**Comment by qwerty287** [comment]
Yes, to prevent such a problem you could set a maximum and it is not possible to change the values so that they are higher that the maximums

**Comment by Mikaela** [comment]
I am missing the `DEFAULT_SHOW_FULL_NAME` (which in GitHub search didn't seem to bring this up) in appearance settings of instances that I don't control.

I view Git forges as a professional context and thus prefer to see the full names when those are set (I thinknthis would also help when viewing an instance from the same sceeen) and there are cases where the username and current name no longer match and can be difficult to update at risk of unforeseen consequences.

For example I have started using my second first name as my calling name while keeping my first first name as an username everywhere, someone may have their surname in username and change it (whether because of marriage or otherwise) and then there is gender transition.

**Comment by lunny** [comment]
I think the first step is to bring them into `system_setting` and next step is to bring it to user setting.
